### ISSUE TITLE
Routines for determining the memory usage

### ISSUE THREAD

**Issue by vsht** [mathics/Mathics#1258]
Mathics currently misses some useful functions that allow to estimate the amount of RAM
available on the current machine and the memory usage. These are: [MemoryAvailable](https://reference.wolfram.com/language/ref/MemoryAvailable.html), [MemoryInUse](https://reference.wolfram.com/language/ref/MemoryInUse.html), [MaxMemoryUsed](https://reference.wolfram.com/language/ref/MaxMemoryUsed.html) and $SystemMemory (apparently not documented, here is the output for my machine with 32 GB RAM. It basically just returns the total amount of RAM in bytes)

```
In[1]:= ?$SystemMemory                                                                                                                       

Out[1]= System`$SystemMemory
        $SystemMemory = 32910770176


        Attributes[$SystemMemory]={Locked, Protected}
```

This is somewhat relevant for #1174 although not critical.

**Comment by rocky** [comment]
The [psutil module](https://pypi.org/project/psutil/) seems to be what would be used in Python to get this information.

**Comment by rocky** [comment]
Should be addressed via #1345. If not, please just reopen this issue with additional details.
